modules/gapi/src/streaming/engine/base_engine.cpp

```diff
@@ -0,0 +1,93 @@
+#include "streaming/engine/base_engine.hpp"
+#include "streaming/vpl/vpl_accel_policy.hpp"
+#include "logger.hpp"
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+VPLProcessingEngine::VPLProcessingEngine(std::unique_ptr<VPLAccelerationPolicy>&& accel) :
+    acceleration_policy(std::move(accel)) {
+}
+
+VPLProcessingEngine::~VPLProcessingEngine() {
+    GAPI_LOG_INFO(nullptr, "destroyed");
+}
+
+VPLProcessingEngine::ExecutionStatus VPLProcessingEngine::process(mfxSession session) {
+    auto sess_it = sessions.find(session);
+    if (sess_it == sessions.end()) {
+
+        // TODO remember the last session status
+        return ExecutionStatus::Processed;
+    }
+
+    session_ptr processing_session = sess_it->second;
+    ExecutionData& exec_data = execution_table[session];
+
+    GAPI_LOG_DEBUG(nullptr, "[" << session <<"] start op id: " << exec_data.op_id);
+    ExecutionStatus status = execute_op(pipeline.at(exec_data.op_id), *processing_session);
+    size_t old_op_id = exec_data.op_id++;
+    if (exec_data.op_id == pipeline.size())
+    {
+        exec_data.op_id = 0;
+    }
+    GAPI_LOG_DEBUG(nullptr, "[" << session <<"] finish op id: " << old_op_id <<
+                                    ", " << processing_session->error_code_to_str() <<
+                                    ", " << VPLProcessingEngine::status_to_string(status) <<
+                                    ", next op id: " << exec_data.op_id);
+
+    if (status == ExecutionStatus::Failed) {
+
+        GAPI_LOG_WARNING(nullptr, "Operation for session: " << session <<
+                                  ", " << VPLProcessingEngine::status_to_string(status) <<
+                                  " - remove it");
+        sessions.erase(sess_it);
+        execution_table.erase(session);
+    }
+
+    if(status == ExecutionStatus::Processed) {
```

Comment:
Why is it a separate if?

Reply:
in comparison with `Failed` it is normal session termination: no need print to log